fix for cmake version 2.8

sc3sc3 opened the pull request:
[CMakeLists.txt](https://github.com/sensestage/hidapi/files/470894/CMakeLists.txt)

sensestage committed:
win: towards a windows version

llloret committed:
win: initial state that compiles

No new features

llloret committed:
win: collection parsing started

llloret committed:
win: element parsing

llloret committed:
win: initial version of element value getting

Not tested yet

llloret committed:
win: get values seems ok

But buttons are not working yet

llloret committed:
win: remove preparsed_data experiment

llloret committed:
win: starting to parse buttons correctly

but need to change the way the elements are allocated. Instead of
allocating everything at the beginning, allocate on the go.

llloret committed:
win: properly create elements and refactor

Taking into account when there is arrange of usages or a HID element,
each should have its own SC element (at least if it is not an array)

llloret committed:
win: some reformatting

llloret committed:
win: some tidying up, and update header file

bagong committed:
Fix policy-warning in cmake output for different cmake versions

llloret committed:
do not the DDK to compile now

The Hid components are part of the standard SDK with MSVC

llloret committed:
fix logical values for buttons, DEBUG_PARSER, C99 loop variable

and some tidying up

bagong committed:
Win: Add extended hidsdi.h for MinGW build

bagong committed:
Win: use added header in MinGW build

bagong committed:
Win: generalize linked setupapi

bagong committed:
Cross: install hut folder

bagong committed:
Cross: change hidapi folder to HID_Support

bagong committed:
Mac: install hut tables into app-bundle

bagong committed:
Provisional fix for install of HUT folder

The install should be defined within SC, and not within the submodule, to allow variables like auxresourcesdir to be used without reassigning them

crucialfelix committed:
Merge pull request #1 from supercollider/hid3.8

HID support for Windows and various additions towards HID in 3.8

bagong commented:
The duplicate line "cmake_policy(SET CMP0048 OLD)" (line 3 and 6) in CMakeLists.txt is still in there. So it is not "a fix for cmake 2.8". Line 6 must go. Sorry.

bagong committed:
Remove duplicate line regarding cmp0048

bagong committed:
Merge pull request #2 from bagong/cmakepol

Remove duplicate line regarding cmp0048

jmckernon committed:
Fix counter bug in hid_parse_input_report()

Initialize the counter variable i to zero in hid_parse_input_report(),
fixing a bug whereby a segfault can happen when the function is entered
more than once.

bagong committed:
Include windows.h to fix MinGW32 build, thanks sonoro1234

tested in mingw- and VS-build

bagong committed:
Merge pull request #3 from jmckernon/topic/hid_parse_input_report_bugfix

Fix counter bug in hid_parse_input_report()

bagong committed:
Merge pull request #4 from supercollider/mingwHandle

Include windows.h to fix MinGW32 build, thanks sonoro1234

awson committed:
Build hidapi against latest mingw-w64 headers (not guarded, perhaps breaks old headers build).

bagong committed:
Add comment to switch to global hidsdi.h

mossheim committed:
Merge pull request #5 from awson/master

Build hidapi against latest mingw-w64 headers

shamazmazum committed:
Fix Findlibusb cmake module for working on FreeBSD

mossheim committed:
Merge pull request #8 from shamazmazum/freebsd-compat

Add FreeBSD compatibility

mossheim commented:
@sonoro1234 @sc3sc3 @bagong , is this repository still being maintained? If not, perhaps we should close this and focus dev work at supercollider/hidapi, which is 33 commits ahead. on a similar note, I have a PR there that I'd appreciate a second look at.

https://github.com/supercollider/hidapi/pull/9

mossheim committed:
cmake: bump to 3.0, use modern project() syntax

mossheim committed:
Merge pull request #10 from brianlheim/topic/policy0048

cmake: bump to 3.0, use modern project() syntax

gurk committed:
Undo "fix hidapi enumeration bug on OSX" - commit c43255b

Caused issues by processing run loop at undesirable moments.

mossheim committed:
Merge pull request #14 from gurk/topic/fix-macos-event-loop-issue

Undo "fix hidapi enumeration bug on OSX" - commit c43255b
